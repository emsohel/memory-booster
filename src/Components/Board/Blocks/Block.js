import React, { Component } from 'react';
import classnames from 'classnames';
import { observer } from 'mobx-react';

class Block extends Component {
  handleClick = () => {
    if (!this.props.board.isPlaying) {
      return;
    }

    this.props.block.selected = true;
    if(!this.props.block.gem === true) {
      //do some error handling
    }
  };

  render() {
    return (
      <div onClick={this.handleClick} className={classnames('box', {
        gem: this.props.block.gem,
        selected: this.props.block.selected
      })}>

      </div>
    );
  }
}

export default observer(['board'])(Block);
