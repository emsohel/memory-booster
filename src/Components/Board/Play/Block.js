import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';

export class Block extends Component {
  static propTypes = {
    width: PropTypes.number.isRequired,
    index: PropTypes.number.isRequired,
    onClick: PropTypes.func.isRequired,
    selected: PropTypes.bool.isRequired
  };

  handleClick = () => {
    this.props.onClick(this.props.index);
  };

  render() {
    const { selected, width } = this.props;
    const classes = classnames('block', { selected });

    return (
      <div style={{ width }} onClick={this.handleClick} className={classes} />
    );
  }
}

export default Block;
