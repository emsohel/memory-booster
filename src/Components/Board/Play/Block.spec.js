import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Block from './Block';
import noop from 'lodash/noop';

it('Play/Block renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render((
    <Block width={50} selected={false} index={1} onClick={noop} />
  ), div);
});

it('Play/Block matches snapshot selected true', () => {
  const component = renderer.create(
    <Block width={50} selected={true} index={1} onClick={noop} />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('Play/Block matches snapshot selected false', () => {
  const component = renderer.create(
    <Block width={50} selected={false} index={1} onClick={noop} />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
