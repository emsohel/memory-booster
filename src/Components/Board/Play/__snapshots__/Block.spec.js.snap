exports[`test Play/Block matches snapshot selected false 1`] = `
<div
  className="box"
  onClick={[Function]}
  style={
    Object {
      "width": 50,
    }
  } />
`;

exports[`test Play/Block matches snapshot selected true 1`] = `
<div
  className="box selected"
  onClick={[Function]}
  style={
    Object {
      "width": 50,
    }
  } />
`;
